src/daft-core/src/array/ops/is_in.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{
+    array::DataArray,
+    datatypes::{
+        BinaryArray, BooleanArray, DaftIntegerType, DaftNumericType, Float32Array, Float64Array,
+        NullArray, Utf8Array,
+    },
+    DataType,
+};
+
+use super::{full::FullNull, DaftIsIn};
+use common_error::DaftResult;
+use std::hash::Hash;
+
+use super::as_arrow::AsArrow;
+
+macro_rules! collect_to_hashset_and_check_membership {
+    ($self:expr, $rhs:expr) => {{
+        let hashset = $rhs
+            .as_arrow()
+            .iter()
+            .collect::<std::collections::HashSet<_>>();
+        let result = $self
+            .as_arrow()
+            .iter()
+            .map(|x| hashset.contains(&x))
+            .collect::<Vec<_>>();
+        Ok(BooleanArray::from(($self.name(), result.as_slice())))
```

Comment:
You can write a `BooleanArray::from_iter` which then doesn't need to materialize the Vec<bool>

https://github.com/Eventual-Inc/Daft/blob/5814922581b48eac48ad7941957e1e9187ce1ae4/src/daft-core/src/array/from_iter.rs#L9